pkg/plugin/vector/store/store.go

```diff
@@ -0,0 +1,70 @@
+package store
+
+import (
+	"context"
+
+	"github.com/grafana/grafana-plugin-sdk-go/backend/log"
+)
+
+type VectorStoreType string
+
+const (
+	VectorStoreTypeQdrant           VectorStoreType = "qdrant"
+	VectorStoreTypeGrafanaVectorAPI VectorStoreType = "grafana/vectorapi"
+)
+
+type Collection struct {
+	Name      string `json:"name"`
+	Dimension int    `json:"dimension"`
+	Model     string `json:"model"`
+}
+
+type SearchResult struct {
+	Payload map[string]any `json:"payload"`
+	Score   float64        `json:"score"`
+}
+
+type ReadVectorStore interface {
+	Collections(ctx context.Context) ([]string, error)
```

Comment:
Do we need this method in `ReadVectorStore`? Could we just have `CollectionExists` (to check if collection exists before calculating embeddings)?

Reply:
Switched to `CollectionExists` for simplicity